base/strings/io.jl

```diff
@@ -210,11 +210,12 @@ IOBuffer(s::SubString{String}) = IOBuffer(view(unsafe_wrap(Vector{UInt8}, s.stri
 # join is implemented using IO
 
 """
-    join(io::IO, strings, delim, [last])
+    join([io::IO,] strings, delim, [last])
 
 Join an array of `strings` into a single string, inserting the given delimiter between
 adjacent strings. If `last` is given, it will be used instead of `delim` between the last
-two strings. For example,
+two strings. If `io` is given, the result is written to `io` rather than returned as
+as a `String`.  For example,
```

Comment:
Unrelated to the PR, but the `For example,` seem kinda redundant with the `# Examples` header.